src/main/scala/beam/router/BeamSkimmer.scala

```diff
@@ -221,16 +239,68 @@ class BeamSkimmer @Inject()(
       event.getServices.getIterationNumber,
       BeamSkimmer.outputFileBaseName + ".csv.gz"
     )
-    FileUtils.writeToFile(
-      filePath,
-      Some(fileHeader),
-      skims
-        .map { keyVal =>
-          s"${keyVal._1._1},${keyVal._1._2},${keyVal._1._3},${keyVal._1._4},${keyVal._2.time},${keyVal._2.cost},${keyVal._2.distance},${keyVal._2.count}"
-        }
-        .mkString("\n"),
-      None
-    )
+    val uniqueModes = skims.map(keyVal => keyVal._1._2).toList.distinct
+    val uniqueTimeBins = (0 to 23)
+
+    beamServicesOpt match {
+      case Some(beamServices) =>
+        val dummyId = Id.create("NA", classOf[BeamVehicleType])
+        val writer = IOUtils.getBufferedWriter(filePath)
+        writer.write(fileHeader)
+        writer.write("\n")
+        beamServices.tazTreeMap.getTAZs
```

Comment:
I'm wondering if a for loop would be more readable here, instead of the heavily nested foreach?